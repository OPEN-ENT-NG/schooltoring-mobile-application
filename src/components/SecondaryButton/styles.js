import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  button: {
    padding: 10,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center"
  },
  buttonText: {
    fontSize: 18,
    color: COLORS.PRIMARY
  }
});
