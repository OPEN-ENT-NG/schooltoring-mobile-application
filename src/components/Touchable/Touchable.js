import React from "react";
import {
  TouchableNativeFeedback,
  TouchableOpacity,
  Platform
} from "react-native";
import PropTypes from "prop-types";

const Touchable = props => {
  return Platform.OS === "android" ? (
    <TouchableNativeFeedback onPress={props.onPress}>
      {props.children}
    </TouchableNativeFeedback>
  ) : (
    <TouchableOpacity onPress={props.onPress}>
      {props.children}
    </TouchableOpacity>
  );
};

Touchable.propTypes = {
  onPress: PropTypes.func.isRequired
};

export default Touchable;
