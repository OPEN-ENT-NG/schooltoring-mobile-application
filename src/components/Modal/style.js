import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  container: {
    width: "100%",
    height: "100%",
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "rgba(10, 10, 10, 0.6)",
    ...StyleSheet.absoluteFillObject
  },
  image: {
    backgroundColor: COLORS.PRIMARY,
    padding: 15
  },
  message: {
    backgroundColor: COLORS.BACKGROUND,
    padding: 10
  },
  title: { color: COLORS.TEXT, margin: 5, fontWeight: "bold" },
  text: { margin: 5 }
});
