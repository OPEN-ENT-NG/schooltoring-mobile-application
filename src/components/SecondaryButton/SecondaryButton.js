import React from "react";
import { Text, View } from "react-native";
import PropTypes from "prop-types";

import Touchable from "../Touchable/Touchable";

import styles from "./styles";
import { COLORS } from "../../styles/common";

const SecondaryButton = props => {
  const children = props.children || (
    <Text style={styles.buttonText}>{props.title}</Text>
  );

  return (
    <View
      style={[
        props.style || {},
        styles.button,
        props.disabled && { backgroundColor: COLORS.GREY }
      ]}
    >
      <Touchable
        disabled={props.disabled}
        onPress={() => props.onPress()}
        hitSlop={{
          top: 10,
          left: 10,
          right: 10,
          bottom: 10
        }}
      >
        {children}
      </Touchable>
    </View>
  );
};

SecondaryButton.propTypes = {
  children: PropTypes.any,
  onPress: PropTypes.func.isRequired,
  style: PropTypes.any,
  title: PropTypes.string,
  disabled: PropTypes.bool
};

export default SecondaryButton;
