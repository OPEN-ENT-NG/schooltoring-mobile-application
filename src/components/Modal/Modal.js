import React from "react";
import { Image, View, Text } from "react-native";
import Proptypes from "prop-types";

import SecondaryButton from "../SecondaryButton/SecondaryButton";
import styles from "./styles";

const Modal = props => (
  <View style={styles.container}>
    <View
      style={{
        width: "70%",
        borderRadius: 5,
        overflow: "hidden"
      }}
    >
      <View style={styles.image}>
        <Image
          style={{ width: "100%" }}
          resizeMode="contain"
          source={props.imageSrc}
        />
      </View>
      <View style={styles.message}>
        <Text id="title" style={styles.title}>
          {props.title}
        </Text>
        <Text id="text" style={styles.text}>
          {props.text}
        </Text>
        <SecondaryButton
          title={props.buttonText || "OK"}
          onPress={props.onPress}
        />
      </View>
    </View>
  </View>
);

export default Modal;

Modal.Proptypes = {
  title: function(props, propName) {
    if (
      props["text"] == true &&
      (props[propName] == undefined || typeof props[propName] != "string")
    ) {
      return new Error("Please provide a title");
    }
  },
  text: function(props, propName) {
    if (
      props["title"] == true &&
      (props[propName] == undefined || typeof props[propName] != "string")
    ) {
      return new Error("Please provide a text");
    }
  },
  imageSrc: Proptypes.string.isRequired,
  buttonText: Proptypes.string,
  onPress: Proptypes.func.isRequired
};
