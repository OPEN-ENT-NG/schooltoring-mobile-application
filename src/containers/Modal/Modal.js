import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

import Modal from "../../components/Modal/Modal";
import { toggleModal } from "../../store/actions/modal";

class ModalContainer extends Component {
  render() {
    if (this.props.isVisible) {
      if (this.props.children) {
        return (
          <Modal
            title={this.props.title}
            text={this.props.text}
            imageSrc={this.props.imageSrc}
          >
            {this.props.children}
          </Modal>
        );
      } else {
        return (
          <Modal
            title={this.props.title}
            text={this.props.text}
            imageSrc={this.props.imageSrc}
            onPress={this.props.toggleModal}
          />
        );
      }
    } else {
      return null;
    }
  }
}

const mapStateToProps = ({ modal }) => ({
  isVisible: modal.isVisible,
  title: modal.title,
  text: modal.text,
  imageSrc: modal.imageSrc,
  children: modal.children
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ toggleModal }, dispatch);
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ModalContainer);
