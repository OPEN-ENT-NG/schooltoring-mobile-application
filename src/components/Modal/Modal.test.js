import React from "react";
import { shallow } from "enzyme";

import Modal from "./Modal";
import SecondaryButton from "../SecondaryButton/SecondaryButton";
import Touchable from "../Touchable/Touchable";

const onPress = jest.fn();
const title = "title";
const text = "text";
const img = require("../../assets/img/send.png");

const rendered = shallow(
  <Modal title={title} text={text} imageSrc={img} onPress={onPress} />
);

describe("Render", () => {
  test("Modal should render without crashing", () => {
    expect(rendered).toMatchSnapshot();
  });

  test("Modal should render with proper title and message", () => {
    const renderedTitle = rendered.find('[id="title"]');
    const renderedText = rendered.find('[id="text"]');

    expect(renderedTitle.contains(title)).toBeTruthy();
    expect(renderedText.contains(text)).toBeTruthy();
  });
});

describe("Interactions", () => {
  test("Modal close button should be called on press", () => {
    const button = rendered
      .find(SecondaryButton)
      .dive()
      .find(Touchable);

    button.prop("onPress")();

    expect(onPress).toHaveBeenCalled();
  });
});
