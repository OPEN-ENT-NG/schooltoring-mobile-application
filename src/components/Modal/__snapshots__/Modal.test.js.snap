// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Modal should render without crashing 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(10, 10, 10, 0.6)",
      "bottom": 0,
      "height": "100%",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Object {
        "borderRadius": 5,
        "overflow": "hidden",
        "width": "70%",
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "#17b6b0",
          "padding": 15,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "backgroundColor": "#fafafa",
          "padding": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        id="title"
        style={
          Object {
            "color": "#232323",
            "fontWeight": "bold",
            "margin": 5,
          }
        }
      >
        title
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        id="text"
        style={
          Object {
            "margin": 5,
          }
        }
      >
        text
      </Text>
      <SecondaryButton
        onPress={[MockFunction]}
        title="OK"
      />
    </Component>
  </Component>
</Component>
`;
